src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -16,141 +16,172 @@
-        proprietaire.setMotDePasse(proprietaireUpdateDTO.getMotDePasse() != null ? hashPassword(proprietaireUpdateDTO.getMotDePasse()) : proprietaire.getMotDePasse());
+    if ((proprietaireUpdateDTO.getEmail() != null && !proprietaireUpdateDTO.getEmail().isBlank())
+        && proprietaireRepository.findFirstByEmail(proprietaireUpdateDTO.getEmail()).isPresent()) {
+      throw new ProprietaireInvalidUpdateBody();
+    }
 
-        return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+    if ((proprietaireUpdateDTO.getNumTel() != null && !proprietaireUpdateDTO.getNumTel().isBlank())
+        && proprietaireRepository.findFirstByNumTel(proprietaireUpdateDTO.getNumTel())
+        .isPresent()) {
+      throw new ProprietaireInvalidUpdateBody();
     }
 
-    @Transactional
-    public ProprietaireDTO deleteProprietaire(@NonNull Long id){
+    Proprietaire proprietaire = optionalProprietaire.get();
+
+    proprietaire.setNom(
+        (proprietaireUpdateDTO.getNom() != null && !proprietaireUpdateDTO.getNom().isBlank())
+            ? proprietaireUpdateDTO.getNom() : proprietaire.getNom());
+    proprietaire.setPrenom(
+        (proprietaireUpdateDTO.getPrenom() != null && !proprietaireUpdateDTO.getPrenom().isBlank())
+            ? proprietaireUpdateDTO.getPrenom() : proprietaire.getPrenom());
+    proprietaire.setEmail(
+        (proprietaireUpdateDTO.getEmail() != null && !proprietaireUpdateDTO.getEmail().isBlank())
+            ? proprietaireUpdateDTO.getEmail() : proprietaire.getEmail());
+    proprietaire.setNumTel(
+        (proprietaireUpdateDTO.getNumTel() != null && !proprietaireUpdateDTO.getNumTel().isBlank())
+            ? proprietaireUpdateDTO.getNumTel() : proprietaire.getNumTel());
+    proprietaire.setMotDePasse(
+        (proprietaireUpdateDTO.getMotDePasse() != null && !proprietaireUpdateDTO.getMotDePasse()
+            .isBlank()) ? hashPassword(proprietaireUpdateDTO.getMotDePasse())
+            : proprietaire.getMotDePasse());
+    proprietaire.setIsPasswordUpdated(
+        proprietaireUpdateDTO.getMotDePasse() != null && !proprietaireUpdateDTO.getMotDePasse()
+            .isBlank() || proprietaire.getIsPasswordUpdated());
+
+    return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+  }
+
+  @Transactional
+  public ProprietaireDTO deleteProprietaire(@NonNull Long id) {
+
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
+    }
 
-        Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
-        if(optionalProprietaire.isEmpty()){
-            throw new ProprietaireNotFoundException();
-        }
+    ProprietaireDTO proprietaireDTO = ProprietaireMapper.entityToDto(optionalProprietaire.get(),
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'proprietaireDTO' must contain no more than '1' consecutive capital letters.